src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/PosixSignalContext.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.InteropServices
+{
+    public sealed class PosixSignalContext
+    {
+        public PosixSignal Signal
+        {
+            get;
+        }
+
+        public bool Cancel
+        {
+            get;
+            set; // TODO: should this throw if Canceling doesn't do anything?
```

Comment:
Do we want this to throw for signals that don't support canceling?

Reply:
Nope. Noop is fine. I think this came up in API review.